src/packet/mod.rs

```diff
@@ -78,9 +84,11 @@ impl PrimitiveValues for ::std::net::Ipv6Addr {
     }
 }
 
+pub mod checksum;
 pub mod ethernet;
 pub mod ip;
 pub mod ipv4;
 pub mod ipv6;
 pub mod udp;
+pub mod tcp;
```

Comment:
I'd rather your work on TCP be separated into a separate pull request - I'll review it here anyway though.